statsmodels/stats/tests/test_cov_tools.py

```diff
@@ -0,0 +1,269 @@
+    g3 = ct.cov_corr_coef_normal(i, j, k, h, corr_r)
+    0.13832670991613805
+    res = 0.1383
+
+    # normal theory variance of correlation coefficient is
+    cr0 = (1-corr_r**2)**2
+    assert_allclose(cr0[1,0], g1, rtol=1e-14)
+    assert_allclose(cr0[1,0], g2, rtol=1e-14)
+
+    cr0low = ct.veclow((1-corr**2)**2)
+    crn = [ct.cov_corr_coef_normal(0, 1, 0, 1, corr),
+           ct.cov_corr_coef_normal(0, 2, 0, 2, corr),
+           ct.cov_corr_coef_normal(1, 2, 1, 2, corr)]
+    assert_allclose(cr0low, crn, atol=1e-14)
+
+    var0, var1, covar, nobs = g1, g2, g3, 20
+    ztest_correlated(diff, var0, var1, covar, nobs)
+    (-2.3642128782748038, 0.018068426888646225)
+    zt1 = -2.30
+
+    # working with indices and masks, e.g. select lower from vec
+    kv = 3
+    order = 'F'
+    ravel_idx_mom4 = np.mgrid[:kv, :kv, :kv, :kv].reshape((4, -1), order=order).T
+    mg2 = np.mgrid[:kv, :kv].reshape((2, -1), order=order).T
+    mask_low = (np.diff(mg2, 1) < 0).squeeze()
+    idx2_dict = dict((tuple(rii.tolist()), ii) for ii, rii in enumerate(mg2))
+
+    covcorr = ct._cov_corr(corr_r, ct._cov_cov(corr_r, 20)*20, 20) * 20
+
+    # extract and individual covariance of correlation coefficients
+    covcorr[idx2_dict[(1, 2)], idx2_dict[(1, 2)]]
+
+
+    # add index or labels to DataFrame
+    import pandas as pd
+    mdlabels = list(map(tuple, mg2[mask_low]))  #convert to list of tuples
+    pd.DataFrame(covcorr[mask_low][:, mask_low], index=mdlabels, columns=mdlabels)
+    """
+              (0, 1)    (0, 2)    (1, 2)
+    (0, 1)  0.170123  0.138327  0.028239
+    (0, 2)  0.138327  0.170123  0.028239
+    (1, 2)  0.028239  0.028239  0.024656
+    """
+
+    cre = ct._cov_corr_elliptical_vech(corr_r, 1, kurt=0)
+    dd = Dup(3)
+    di = np.linalg.pinv(dd)
+    #(di.dot(cre).dot(di.T))[np.array([[1,2,4]]).T, np.array([[1,2,4]])] / covcorr[mask_low][:, mask_low]
+    #print(cre[np.array([[1,2,4]]).T, np.array([[1,2,4]])] / covcorr[mask_low][:, mask_low])
+    # Note create index arrays
+
+    # selector masks vor vec, vech and veclow
+    mask_vech = (np.diff(mg2, 1) <= 0).squeeze()
+    mg2vech = mg2[mask_vech]
+    mask_vech2veclow = (np.diff(mg2vech, 1) < 0).squeeze()
+    cr1 = cre[np.ix_(mask_vech2veclow, mask_vech2veclow)]
+    cr2 = di.dot(ct._cov_corr(corr_r, ct._cov_cov(corr_r, 1), 1)).dot(di.T)[np.ix_(mask_vech2veclow, mask_vech2veclow)]
+    vr1 = ((1 -vech(corr_r)**2)**2)[mask_vech2veclow]
+    assert_allclose(cr1.diagonal(), vr1, rtol=1e-14)
+    # same as
+    # Note we need to set nobs=1 in call to _cov_cov  Bug or Definition?
+    cr3 = ct._cov_corr(corr_r, ct._cov_cov(corr_r, 1), 1)[np.ix_(mask_low, mask_low)]
+    # same as
+    cr4 = ct._cov_corr_elliptical_vech(corr_r, 1, kurt=0, drop=True)
+
+    n = 3
+    n_low = n * (n - 1) / 2
+    assert_equal(cr1.shape, (n_low, n_low))
+    assert_allclose(cr2, cr1, rtol=1e-14)
+    assert_allclose(cr3, cr1, rtol=1e-14)
+    assert_allclose(cr4, cr1, rtol=1e-14)
```

Comment:
there are two underlying cov_corr functions, the other variation is on selection and selection options to get to minimal, unique correlations
missing comparison to steiger hakstian example, single element function and reference numbers, only tested for normal case, elliptical with small kurtosis should be close. _cov_corr works for arbitrary cov_cov but scaling of mom4 is not checked (1/nobs) for consistency.